dataclass_wizard/abstractions.py

```diff
@@ -88,7 +88,7 @@ def list_to_json(cls: Type[W],
 
 
 @dataclass
-class AbstractParser(ABC):
+class AbstractParser(ABC, Generic[T, TT]):
```

Comment:
I thought about introducing an intermediate layer for vast majority of the cases, that would be specified with T only.